h2o-algos/src/main/java/hex/deepwater/DeepWater.java

```diff
@@ -69,7 +69,7 @@ static boolean haveBackend(DeepWaterParameters.Backend b) {
 
   @Override public boolean isSupervised() { return !_parms._autoencoder; }
 
-  @Override protected int nModelsInParallel() { return 1; }
+  @Override protected int nModelsInParallel(int folds) { return 1; }
```

Comment:
Do we need this new argument? Can it be retrieved from model parameters instead?

Reply:
it's the result of `ModelBuilder#nFoldWork()` that is computed during xval,  so as it's a protected method only, it's safer to pass this result as a parameter.

Reply:
Can we still keep nModelsInParallel in the no-parameter version as well and mark it as deprecated? There are internal projects that depend on it -> people out there might be using it as well.

The method could produce a warning and the behavior doesn't have to be perfect. WDYT?

Reply:
@michalkurka sure, I need to hook it into the one with param though as it's the last one that is used by `ModelBuilder`
 
change proposal in ModelBuilder:
```
@Deprecated protected int nModelsInParallel() { return 0; }

protected int nModelsInParallel(int folds) { 
   if (nModelsInParallel() > 0) return nModelsInParallel();
   // current logic goes there
}

Reply:
when you say internal projects are using it: do you mean they're overriding it? or are they consuming/calling it?
my proposal works in the former case but not the later.

Reply:
in case or consuming the default impl would rather be sth like:
```
@Deprecated protected int nModelsInParallel() {
  return nModelsInParallel(_parms._nfolds)
}
```

Reply:
I think your proposal makes sense

Reply:
@michalkurka which one? :)
the first one right? I expect this being mainly overridden, not called internally as it seems to have been introduced for xval only.

Reply:
I don't see the use case for calling directly, but overriding the method in a custom implementation makes sense for me (we do it as well in the code).

Reply:
@michalkurka just added it to `ModelBuilder`: see https://github.com/h2oai/h2o-3/pull/3218/files#diff-ab9d2bd762456be481b3fa49912d6b64R391